frame/support/procedural/src/pallet/expand/tt_default_parts.rs

```diff
@@ -90,9 +92,8 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 		macro_rules! #default_parts_unique_id {
 			{
 				$caller:tt
-				frame_support = [{ $($frame_support:ident)::* }]
 			} => {
-				$($frame_support)*::tt_return! {
+				#tt_return_path! {
```

Comment:
this doesn't work, except if force frame crate to be in scope. IMO we should still use some path provided by the caller, instead of frame_support path we could ask for path_to_tt_return_path also.

Reply:
Here path to `frame_support` is given by `construct_runtime`.
if `construct_runtime` is used from `frame_support` and not `frame` then it needs to give the path to `frame_support` from `frame_support` i.e. `crate_name(frame-support)`
if `construct_runtime` is used from `frame` and not `frame_support` then it needs to give the path to `frame_support` from `frame` as `crate_name(frame)::deps::frame_support`

so in construct_runtime we should query `crate_name("frame")` if it resolve all good then we give `crate_name(frame)::deps::frame_support` if it doesn't resolve then we query `crate_name(frame-support)` if the latter resolve then good we give  `crate_name(frame-support)` as path for frame_support and if it doesn't resolve then we return an error

Reply:
Okay, I see the point about needing to distinguish between the cases where `crate_name("frame")` is okay and when `crate_name("frame_support")`.


For now, I seem to have another issue. Let's assume both the pallet and runtime are built with `frame`. If something like `frame::deps::frame_support` is passed in here, it is not received well in the receiver macro. And I cannot seem to get it to work with any configuration. Ideally, I was hoping `frame_support = [{ $frame_support:path }]` would work.

Reply:
Your idea of passing in the full path to `tt_return` is golden and seem to have worked 💯

Reply:
so the issue is solved no ?
currently it should already accept a path (more precisely an serie of ident and `::`):
```
				frame_support = [{ $($frame_support:ident)::* }]
```

Reply:
Yeah, solved now.